airflow/providers/amazon/aws/utils/connection_wrapper.py

```diff
@@ -127,9 +127,18 @@ def __post_init__(self, conn: "Connection"):
         self.password = conn.password
         self.extra_config = deepcopy(conn.extra_dejson)
 
-        if self.conn_type != "aws":
+        if self.conn_type.lower() == "s3":
             warnings.warn(
-                f"{self.conn_repr} expected connection type 'aws', got {self.conn_type!r}.",
+                f"{self.conn_repr} has connection type 's3', which is deprecated. "
+                "Please update your connection and set `conn_type='aws'`.",
```

Comment:
One thing .... it seems that we may cause some confusion or unnecessary alarm with this.  Because.... nothing about this connection needs to be changed except the conn_type ...  Seems maybe it would be worth adding a sentence emphasizing that somehow?  what do you think?